service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackMessageListenerClient.java

```diff
@@ -0,0 +1,108 @@
+     * if the notification should be received by a different instance.
+     *
+     * @param connectionString The connection string for the IotHub
+     * @param iotHubServiceClientProtocol protocol to use
+     * @param feedbackBatchMessageCallback the callback to be executed whenever a feedback message is received
+     */
+    public FeedbackMessageListenerClient(String connectionString, IotHubServiceClientProtocol iotHubServiceClientProtocol, FeedbackBatchMessageCallback feedbackBatchMessageCallback)
+    {
+        Tools.throwIfNullOrEmpty(connectionString, "iotHubConnectionString cannot be null");
+        Tools.throwIfNull(iotHubServiceClientProtocol, "iotHubServiceClientProtocol cannot be null");
+
+        try
+        {
+            this.iotHubConnectionString = IotHubConnectionStringBuilder.createConnectionString(connectionString);
+        }
+        catch (IOException e)
+        {
+            //No IO operation is occuring during parsing of the connection string, so the IOException is just confusing. Wrapping
+            // it in an IllegalArgumentException instead so that customers don't have to catch an IOException
+            throw new IllegalArgumentException("Failed to build iot hub connection string", e);
+        }
+
+        this.sasTokenExpiryTime = DEFAULT_SAS_TOKEN_EXPIRY_TIME;
+        this.iotHubServiceClientProtocol = iotHubServiceClientProtocol;
+
+        this.hostName = this.iotHubConnectionString.getHostName();
+        this.userName = this.iotHubConnectionString.getUserString();
+
+        Tools.throwIfNullOrEmpty(this.hostName, "The provided connection string is missing a hostName");
+        Tools.throwIfNullOrEmpty(this.hostName, "The provided connection string is missing a userName");
+        amqpReceiveHandler = new AmqpFeedbackListenerHandler(this.hostName, this.userName, this.iotHubServiceClientProtocol, feedbackBatchMessageCallback);
+    }
+
+    /**
+     * Open the AMQP connection, and start listening for feedback messages. Any feedback messages that arrive
+     * will be given to the user through the FileUploadNotificationCallback provided in the constructor
+     *
+     * @throws IOException If opening the connection fails
+     * @throws InterruptedException If the thread gets interrupted while waiting for the connection to open
+     */
+    public synchronized void open() throws IOException, InterruptedException
+    {
+        amqpReceiveHandler.open(new IotHubServiceSasToken(iotHubConnectionString, sasTokenExpiryTime).toString());
+        log.info("Feedback message listener connection opened");
+    }
+
+    /**
+     * Close the AMQP connection. No feedback messages will be received on this listener until it is opened again.
+     * Closed instances may be re-opened and re-closed any number of times
+     */
+    public synchronized void close()
+    {
+        amqpReceiveHandler.close();
```

Comment:
```amqpReceiveHandler``` is instantiated in the constructor, and never set to null. I don't see how it could ever be null, so I'd rather not add a check here